compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -1523,6 +1523,70 @@ void Resolver::exit(const uast::New* node) {
   }
 }
 
+bool Resolver::enter(const For* loop) {
+  enterScope(loop);
+
+  const AstNode* iterand = loop->iterand();
+  iterand->traverse(*this);
+
+  ResolvedExpression& iterandRE = byPostorder.byAst(iterand);
+
+  auto& MSC = iterandRE.mostSpecific();
+  bool isIter = MSC.isEmpty() == false &&
+                MSC.numBest() == 1 &&
+                MSC.only()->untyped()->kind() == uast::Function::Kind::ITER;
+
+  bool wasResolved = iterandRE.type().isUnknown() == false &&
+                     iterandRE.type().isErroneousType() == false;
+
+  QualifiedType idxType = QualifiedType(QualifiedType::UNKNOWN,
+                                        ErroneousType::get(context));
```

Comment:
I don't know if it is moot or not for performance, but I'd probably leave it uninitialized here and set it to this only in the error case